CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+# Contributing
+
+You can contribute to Silk.NET with issues and PRs. Simply filing issues for problems you encounter is a great way to contribute. Contributing code is also greatly appreciated.
+
+## Considerations
+The Silk.NET team will merge changes that improve the library significantly. We will not merge changes that have narrowly-defined benefits, or are breaking in some way. All contributions must also follow all other guidelines outlined in this document.
+
+## DOs and DON'Ts
+Please:
+- **DO** talk to us in #team-talk in the Discord or open a discussion issue if your contribution is sizeable.
+- **DO** keep the discussions around contributions focused. If you have another matter to discuss, rather than creating a massive tangent in the current discussion, open up a new one.
+- **DO** follow the code of conduct if discussing on GitHub and/or the Discord #rules if discussing on Discord.
+- **DON'T** make PRs that don't directly affect the end user, such as style changes. These are best done as part of a PR related to the area in question. Documentation is fine (and encouraged!), as this is useful to the end user.
+- **DON'T** commit code you didn't write without following its license and following our guidelines for doing so in the Contributor License Agreement. If you are unable to license some code, don't commit it.
+- **DON'T** surprise us with big pull requests or big API changes without talking to us first!
+- **DON'T** make PRs for legal or administrative documents, such as the license, file headers, or code of conduct. If you see something off, let us know and we'll change it.
+
+## Breaking changes
+Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected. There are exceptions to this, such as if the contribution is made against a major/breaking version branch (such as `2.0` instead of `master`), however this is subject to the team's approval. At risk of sounding like a broken record, talk to us about your idea first!
```

Comment:
#265 really needs to happen for this to be verifiable.
Also, we should probably add that, while binder changes should keep signatures compatible, sometimes upstream changes happen while one is working on the binder.
Imo, in that case it's best to add a comment to the PR, and make a separate PR to update bindings using the current version of the binder.
That way it will can just check that no changes were made to the API Surface, without checking for each check whether it really is an upstream change (which would be error prone)

Reply:
```suggestion
Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected. There are exceptions to this, such as if the contribution is made against a major/breaking version branch (such as `2.0` instead of `master`), however this is subject to the team's approval. At risk of sounding like a broken record, talk to us about your idea first!

### Generated bindings
Sometimes upstream specification changes happen while working on your contribution. Unfortunately, we are not in control of breaking changes created by the upstream spec sources, and if one happens while working on your contribution it is not a cause for concern.

If your contribution affects binder output, let us know by commenting on the pull request. If the change isn't as a result of your modifications to the BuildTools (if any), it is recommended that you don't commit the .gen.cs files and instead do this in a separate PR.
```